monkestation/code/modules/ghost_critters/client_addons.dm

```diff
@@ -0,0 +1,78 @@
+/client
+	var/ghost_critter_cooldown = 0
+
+
+/client/proc/get_critter_spawn(obj/structure/ghost_critter_spawn/spawner)
+	var/list/basic_list = list(
+		/mob/living/basic/mouse,
+		/mob/living/basic/axolotl,
+		/mob/living/basic/butterfly,
+		/mob/living/basic/crab,
+		/mob/living/basic/mothroach
+	)
+
+	var/list/mobs_to_pick = list()
+
+	mobs_to_pick += return_donator_mobs()
+
+	if(!patreon.has_access(ACCESS_ASSISTANT_RANK) && !is_admin(src) && !length(mobs_to_pick))
+		return pick(basic_list)
+
+	mobs_to_pick += basic_list
+
+	var/list/spawned_mobs = list()
+	var/list/deletors = list()
+	for(var/mob/living/basic/basic as anything in mobs_to_pick)
+		var/mob/living/basic/created = new basic()
+		spawned_mobs += list(created.name = created)
+		deletors += created
+
+	var/choice = show_radial_menu(mob, spawner, spawned_mobs, tooltips = TRUE)
+	if(!choice)
+		spawned_mobs = null
+		QDEL_LIST(deletors)
+		return pick(basic_list)
+	var/mob/living/basic/picked = spawned_mobs[choice]
+	var/mob_type = picked.type
+	spawned_mobs = null
+	QDEL_LIST(deletors)
+	return mob_type
+
+/client/proc/try_critter_spawn(obj/structure/ghost_critter_spawn/spawner)
+	var/turf/open/turf = get_turf(spawner)
+
+	var/mob/living/basic/spawned_mob = get_critter_spawn(spawner)
+	var/mob/living/basic/created_mob = new spawned_mob(turf)
+
+	var/cooldown_tiem = get_critter_cooldown()
+	ghost_critter_cooldown = world.time + 15 MINUTES
+
+	if(!mob.mind)
+		mob.mind = new /datum/mind(key)
+
+	mob.mind.transfer_to(created_mob, TRUE)
```

Comment:
if this is taking from an observer then this needs to do a key transfer instead as this keeps antag status from previous lives